keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -17,10 +17,13 @@
  */
 package com.netflix.spinnaker.keel.api
 
+import com.fasterxml.jackson.annotation.JacksonInject
+
 /**
  * A resource spec which is located in an account and one or more regions.
  */
 interface Locatable<T : Locations<*>> : ResourceSpec {
+  @get:JacksonInject("locations")
   val locations: T
 }
```

Comment:
Adding this `@JacksonInject` annotation means that if there is no corresponding data in the JSON being parsed, Jackson will look in the deserialization context for "injected" values using this key.